File: android-bridgeUI/src/main/java/me/yamlee/bridge/ui/dialog/internel/SimpleEditDialog.java
package me.yamlee.bridge.ui.dialog.internel;

import android.app.Dialog;
import android.content.Context;
import android.text.InputFilter;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import me.yamlee.bridge.ui.R;
import timber.log.Timber;

/**
 * 简单的带编辑提示框,基本样式为:标题,输入框,确认按钮
 *
 * @author yamlee
 */
public class SimpleEditDialog extends Dialog {
    private Context mContext;

    public SimpleEditDialog(Context context) {
        super(context);
    }

    public SimpleEditDialog(Context context, int themeResId) {
        super(context, themeResId);
    }

    protected SimpleEditDialog(Context context, boolean cancelable, OnCancelListener cancelListener) {
        super(context, cancelable, cancelListener);
    }

    public void setEditMsg(String msg) {
        try {
            EditText editText = (EditText) findViewById(R.id.et_dialog_simple_edit_content);
            if (editText != null) {
                if (!TextUtils.isEmpty(msg)) {
                    editText.setText(msg);
                    editText.setSelection(msg.length());
                } else {
                    editText.setText("");

                }
            }
        } catch (IndexOutOfBoundsException e) {
            //如果设置的edit最大长度setSelection可能会包indexOutOfBound异常
            Timber.e(e);
        }
    }

    public void setDialogTitlle(String msg) {
        TextView title = (TextView) findViewById(R.id.tv_dialog_simple_edit_title);
        if (title != null) {
            if (!TextUtils.isEmpty(msg)) {
                title.setText(msg);
            }
        }
    }

    private void initView(Context context) {
        mContext = context;
    }

    public static SimpleEditDialog.Builder builder() {
        return new Builder();
    }

    public static class Builder {
        private String title;
        private String confirmBtnText = "";
        private String editTextHint = "";
        private boolean canTouchOutDismiss;
        SimpleEditDialog mDialog = null;
        private int maxLength;
        private OnConfirmClickListener confirmClickListener;


        public Builder setTouchOutDismiss(boolean canTouchOutDismiss) {
            this.canTouchOutDismiss = canTouchOutDismiss;
            return this;
        }

        public Builder setEditInputMaxLength(int length) {
            this.maxLength = length;
            return this;
        }

        public Builder setTitle(String title) {
            this.title = title;
            return this;
        }

        public Builder setEditHint(String hintMsg) {
            this.editTextHint = hintMsg;
            return this;
        }


        public Builder setConfirmBtnText(String confirmBtnText) {
            this.confirmBtnText = confirmBtnText;
            return this;

        }

        public Builder setConfirmClickListener(OnConfirmClickListener confirmClickListener) {
            this.confirmClickListener = confirmClickListener;
            return this;
        }

        public SimpleEditDialog build(Context context) {

            TextView tvTitle, tvConfirm;

            View mDialogView;
            LayoutInflater inflater = (LayoutInflater)
                    context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            mDialog = new SimpleEditDialog(context, R.style.publish_dialog);
            mDialogView = inflater.inflate(
                    R.layout.view_dialog_simple_edit, null);
            final EditText editText = (EditText) mDialogView.findViewById(R.id.et_dialog_simple_edit_content);
            if (maxLength != 0) {
                InputFilter[] inputFilters = new InputFilter[1];
                inputFilters[0] = new InputFilter.LengthFilter(maxLength);
                editText.setFilters(inputFilters);
            }
            if (!TextUtils.isEmpty(editTextHint)) {
                editText.setHint(editTextHint);
            }
            tvTitle = (TextView) mDialogView.findViewById(R.id.tv_dialog_simple_edit_title);
            if (TextUtils.isEmpty(title)) {
                title = context.getString(R.string.common_dialog_title);
            }
            tvTitle.setText(title);

            tvConfirm = (TextView) mDialogView.findViewById(R.id.tv_dialog_simple_edit_confirm);
            if (TextUtils.isEmpty(confirmBtnText)) {
                confirmBtnText = context.getString(R.string.confirm);
            }
            tvConfirm.setText(confirmBtnText);
            tvConfirm.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (confirmClickListener != null) {
                        confirmClickListener.onConfirm(editText.getText().toString());
                    }
                }
            });

            mDialog.setContentView(mDialogView);
            mDialog.setCanceledOnTouchOutside(canTouchOutDismiss);
            return mDialog;
        }

        public interface OnConfirmClickListener {
            void onConfirm(String editMsg);
        }
    }

}

File: android-jsBridgeAll/src/main/java/me/yamlee/bridge/ui/dialog/internel/SimpleEditDialog.kt
package me.yamlee.bridge.ui.dialog.internel

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.text.InputFilter
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.widget.EditText
import android.widget.TextView

import me.yamlee.bridge.ui.R
import timber.log.Timber

/**
 * 简单的带编辑提示框,基本样式为:标题,输入框,确认按钮
 *
 * @author yamlee
 */
class SimpleEditDialog : Dialog {
    private var mContext: Context? = null

    constructor(context: Context) : super(context) {}

    constructor(context: Context, themeResId: Int) : super(context, themeResId) {}

    protected constructor(context: Context, cancelable: Boolean, cancelListener: DialogInterface.OnCancelListener) : super(context, cancelable, cancelListener) {}

    fun setEditMsg(msg: String) {
        try {
            val editText = findViewById<EditText>(R.id.et_dialog_simple_edit_content)
            if (editText != null) {
                if (!TextUtils.isEmpty(msg)) {
                    editText.setText(msg)
                    editText.setSelection(msg.length)
                } else {
                    editText.setText("")

                }
            }
        } catch (e: IndexOutOfBoundsException) {
            //如果设置的edit最大长度setSelection可能会包indexOutOfBound异常
            Timber.e(e)
        }

    }

    fun setDialogTitlle(msg: String) {
        val title = findViewById<TextView>(R.id.tv_dialog_simple_edit_title)
        if (title != null) {
            if (!TextUtils.isEmpty(msg)) {
                title.text = msg
            }
        }
    }

    private fun initView(context: Context) {
        mContext = context
    }

    class Builder {
        private var title: String? = null
        private var confirmBtnText = ""
        private var editTextHint = ""
        private var canTouchOutDismiss: Boolean = false
        internal var mDialog: SimpleEditDialog? = null
        private var maxLength: Int = 0
        private var confirmClickListener: OnConfirmClickListener? = null


        fun setTouchOutDismiss(canTouchOutDismiss: Boolean): Builder {
            this.canTouchOutDismiss = canTouchOutDismiss
            return this
        }

        fun setEditInputMaxLength(length: Int): Builder {
            this.maxLength = length
            return this
        }

        fun setTitle(title: String): Builder {
            this.title = title
            return this
        }

        fun setEditHint(hintMsg: String): Builder {
            this.editTextHint = hintMsg
            return this
        }


        fun setConfirmBtnText(confirmBtnText: String): Builder {
            this.confirmBtnText = confirmBtnText
            return this

        }

        fun setConfirmClickListener(confirmClickListener: OnConfirmClickListener): Builder {
            this.confirmClickListener = confirmClickListener
            return this
        }

        fun build(context: Context): SimpleEditDialog {

            val tvTitle: TextView
            val tvConfirm: TextView

            val mDialogView: View
            val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            mDialog = SimpleEditDialog(context, R.style.publish_dialog)
            mDialogView = inflater.inflate(
                    R.layout.view_dialog_simple_edit, null)
            val editText = mDialogView.findViewById<View>(R.id.et_dialog_simple_edit_content) as EditText
            if (maxLength != 0) {
                val inputFilters = arrayOfNulls<InputFilter>(1)
                inputFilters[0] = InputFilter.LengthFilter(maxLength)
                editText.filters = inputFilters
            }
            if (!TextUtils.isEmpty(editTextHint)) {
                editText.hint = editTextHint
            }
            tvTitle = mDialogView.findViewById<View>(R.id.tv_dialog_simple_edit_title) as TextView
            if (TextUtils.isEmpty(title)) {
                title = context.getString(R.string.common_dialog_title)
            }
            tvTitle.text = title

            tvConfirm = mDialogView.findViewById<View>(R.id.tv_dialog_simple_edit_confirm) as TextView
            if (TextUtils.isEmpty(confirmBtnText)) {
                confirmBtnText = context.getString(R.string.confirm)
            }
            tvConfirm.text = confirmBtnText
            tvConfirm.setOnClickListener {
                if (confirmClickListener != null) {
                    confirmClickListener!!.onConfirm(editText.text.toString())
                }
            }

            mDialog!!.setContentView(mDialogView)
            mDialog!!.setCanceledOnTouchOutside(canTouchOutDismiss)
            return mDialog!!
        }

        interface OnConfirmClickListener {
            fun onConfirm(editMsg: String)
        }
    }

    companion object {

        fun builder(): SimpleEditDialog.Builder {
            return Builder()
        }
    }

}
